packages/gatsby/src/utils/validate-page-component.ts

```diff
@@ -74,26 +74,25 @@ export function validatePageComponent(
       }
     }
 
-    const includesDefaultExport =
-      fileContent.includes(`export default`) ||
-      fileContent.includes(`module.exports`) ||
-      fileContent.includes(`exports.default`) ||
-      fileContent.includes(`exports["default"]`) ||
-      fileContent.match(/export \{.* as default.*\}/s)
-
     // this check only applies to js and ts, not mdx
-    if (
-      /\.(jsx?|tsx?)/.test(path.extname(component)) &&
-      !includesDefaultExport
-    ) {
-      return {
-        error: {
-          id: `11328`,
-          context: {
-            component,
+    if (/\.(jsx?|tsx?)/.test(path.extname(component))) {
```

Comment:
Can't we do this instead? I'm allergic to regex checks when efficient and simpler string comparisons suffice.

```suggestion
    const ext = path.extname(component);
    if (ext.endsWith('.jsx') || ext.endsWith('.tsx')) {
```
We can probably even drop the `path.extname()` call because we're only interested in the last bit.

Reply:
But then also need test for `.js` and `.ts` to match the regex

Reply:
Can this also done like (the I assume that `path.extname` gives the last part):

```JavaScript 
[`.js`,`.jsx`,`.ts`,`.tsx`].includes(ext)
```

Reply:
Good point. Yeah, I think so :shrug:

Reply:
```suggestion
    if ([`.js`,`.jsx`,`.ts`,`.tsx`].includes(path.extname(component))) {
```

@pvdz PR welcome?

Reply:
open PR:

- #24843 `chore(gatsby): validate-page-component - simplify regex`